sycl/doc/design/PropertySets.md

```diff
@@ -0,0 +1,245 @@
+# SYCL binary property sets
+
+To communicate information about SYCL binaries to the SYCL runtime, the
+implementation produces sets of properties. The intention of this design
+document is to describe the structure of the property sets and define the
+representation and meaning of pre-defined property set names.
+
+## Property sets structure
+
+A property set consists of a reserved name, enclosed in square brackets,
+followed by a series of string key and value pairs. The set name and each entry
+in the set are separated by a newline.
+
+The string key and value pairs have the following format:
+```
+<string key>=<value type>|<value>
+```
+
+The value type is a string and the value of it has the following meaning for the
+corresponding value:
+
+| Value type | Description                               |
+| ---------- | ----------------------------------------- |
+| "0"        | The value has no known type.              |
```

Comment:
This seems underspecified.  However, I don't see any property that uses value type "0".  Should we remove this from the spec?

Reply:
This is generally considered an invalid case, but it is named in the implementation. Do you prefer it says that or that I remove the entry?